### DISCUSSION TITLE
DI bindings generated at compile time

### DISCUSSION THREAD

**Discussion by arsentiikaptsan** [activej/activej#170] [Q&A]
Hi!

I wonder if activej has plans to make an annotation processor that would generate DI bindings at compile time, similar to how dagger2 or Micronaut DI work? This way we can eliminate wiring at start up via reflexion. Thanks!

**Comment by eduard-vasinskyi** [comment]
Hello, @arsentiikaptsan

By design, ActiveJ Inject has powerful facilities for introspection,  analysis, generation, and transformation of binding in runtime, it’s not just static bindings that rely on annotations. In fact, DI annotations are merely a syntactic sugar and are completely optional to use.

Moreover, ActiveJ Inject allows eliminating reflection overhead altogether by wiring components using [ModuleBuilder](https://github.com/activej/activej/blob/master/core-inject/src/main/java/io/activej/inject/module/ModuleBuilder.java) DSL, producing fast optimized bindings for runtime.

**Comment by arsentiikaptsan** [comment]
I got your point, thank you!
